packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -603,29 +615,91 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
             }
 
             const nodeText = tr.doc.textBetween(pos, pos + node.nodeSize, undefined, ' ');
-            this.findAutoLinks(nodeText)
+
+            const foundAutoLinks: FoundAutoLink[] = [];
+
+            for (const match of findMatches(nodeText, NON_WHITESPACE_REGEX)) {
+              const matchedText = getMatchString(match);
+
+              // Skip if URL doesn't have minimum length
+              // TODO: Could make this configurable
+              if (matchedText.length < MIN_LINK_LENGTH) {
+                continue;
+              }
+
+              const matchStart = match.index;
+              const matchEnd = matchStart + matchedText.length;
+              const urlInMatchedText = this.findURLInString(matchedText);
+              const linkMarkInRange = this.getLinkMarksInRange(
+                doc,
+                pos + matchStart,
+                pos + matchEnd,
+                true,
+              )[0];
+              const href = (urlInMatchedText && this.buildHref(urlInMatchedText.url)) || '';
+              const positionStart = pos + matchStart + (urlInMatchedText?.startIndex || 0);
+              const positionEnd = positionStart + href.length;
+
+              const unbalancedLinkUpdated = urlInMatchedText
+                ? linkMarkInRange &&
+                  !isBalanced(linkMarkInRange.text) &&
+                  isBalanced(linkMarkInRange.mark.attrs.href)
+                : false;
+
+              const outsideSelectionRange = urlInMatchedText
+                ? positionEnd < to + (from === to ? 1 : -1) ||
+                  positionStart > from - (from === to ? 1 : 0) ||
+                  (linkMarkInRange &&
+                    (linkMarkInRange.to >= to ||
+                      !within(from, linkMarkInRange.from, linkMarkInRange.to)))
+                : false;
+
+              const isExistingLinkTextNotEqualHref =
+                urlInMatchedText && linkMarkInRange
```

Comment:
As we determined in a previously merged pr #1741 link text and "href" attribute should be equal otherwise a link is not valid.

If not valid the link is removed on line 670.